src/Strategy/Go/GoModGraph.hs

```diff
@@ -179,16 +192,36 @@ analyze dir = do
   goModGraphStdout <- context ("Getting selected dependencies versions using, " <> toText (show goModGraphCmd)) $ execParser parseGoModGraph dir goModGraphCmd
   ggm <- fromMaybeText "expected to find main module, but found no main module!" mainMod
 
-  let graph = buildGraph goModGraphStdout ggm selectedMods directMods filterModsNotUsedInBuild
+  let graph = buildGraph goModGraphStdout ggm selectedMods directMods filterModsNotUsedInBuild modReplacements
   pure (graph, Complete)
   where
+    -- TODO: need to convert the version from the replacement into something well-typed
     toGoGraphMod :: GoListModule -> GoGraphMod
     toGoGraphMod GoListModule{..} = case (toGoModVersion =<< version) of
       Nothing -> MainMod path
       Just pv -> OtherMod path pv
 
-    withoutMain :: [GoListModule] -> Set GoGraphMod
-    withoutMain = fromList . map toGoGraphMod . filter (not . isMain)
+    toGraphModWithReplacement :: GoListModule -> (GoGraphMod, Maybe ModWithReplacement)
```

Comment:
[Prefer] That we order definition of function in order they appear for easy readability (so we don't have to scan back and forth when reading)

E.g 
```haskell
    findSelectedMods :: [GoListModule] -> (Set GoGraphMod, GoModReplacements)
    findSelectedMods =
      bimap fromList (mkReplacementMap . catMaybes)
        . unzip
        . map toGraphModWithReplacement
        . filter (not . isMain)
 
     mkReplacementMap :: [ModWithReplacement] -> GoModReplacements
    mkReplacementMap = GoModReplacements . Map.fromList . map unModWithReplacement
 
     toGraphModWithReplacement :: GoListModule -> (GoGraphMod, Maybe ModWithReplacement)
     toGraphModWithReplacement listMod =
      case toGoGraphMod listMod of
        m@(MainMod _) -> (m, Nothing)
        m@(OtherMod _ _) ->
          ( m
          , do
              GoModuleReplacement{..} <- moduleReplacement listMod
              version <- toGoModVersion versionReplacement
              Just $ ModWithReplacement (m, OtherMod pathReplacement version)
       
```

Reply:
I noticed this as well and that might be good. One other option I thought of was trying to make wrapper types for each of the different kinds of GoGraphMod sets. At least then even with a function with so many arguments the type system would help make sure they're put in the right order. I will hack on that or try to make an argument record and push those changes.

Reply:
I've done this